CurlWrapper.php

```diff
@@ -57,6 +57,10 @@ class CurlWrapper
      * @var array cURL transfer info
      */
     protected $transferInfo = array();
+    /**
+     * @var string cURL response header data
+     */
+    protected $responseHeaders = '';
```

Comment:
So this is string, but after parsing it's becoming array. We should use 2 variables here (for raw headers and for parsed headers) and I explain below why.